src/vmm_core.ml

```diff
@@ -140,23 +184,17 @@ module Name = struct
   let to_list (p, name) =
     Option.fold ~none:p ~some:(fun n -> p @ [ n ]) name
 
+  let to_labels (p, name) =
+    Option.fold ~none:p ~some:(fun n -> p @ [ n ]) name
+
   let of_list ids =
-    match path_of_list ids with
+    match Path.of_strings ids with
     | Error _ -> Error (`Msg "invalid name")
     | Ok _ -> of_path ids
 
-  let path_to_string x = String.concat ":" x
-
-  let path_of_string str =
-    let ps = String.split_on_char ':' str in
-    let ps = match ps with | ""::tl -> tl | ps -> ps in
-    if List.for_all valid_label ps then
-      Ok ps
-    else
-      Error (`Msg "invalid path")
-
   let to_string (p, n) =
-    path_to_string p ^ ":" ^ Option.value ~default:"" n
+    Path.to_string p ^ ":" ^
+    Option.value ~default:"" (Option.map Label.to_string  n)
```

Comment:
I think here it's not necessary to do `Label.to_string` because `Label.t` isn't opaque here. But it's fine.